pipenv/utils/resolver.py

```diff
@@ -447,9 +447,49 @@ def resolve(self):
                 self.resolved_tree.update(self.results)
         return self.resolved_tree
 
+    def _get_pipfile_markers(self, pipfile_entry):
+        sys_platform = pipfile_entry.get("sys_platform")
+        platform_machine = pipfile_entry.get("platform_machine")
+        markers = pipfile_entry.get("markers")
+
+        if sys_platform:
+            sys_platform = f"sys_platform {sys_platform}"
+        if platform_machine:
+            platform_machine = f"platform_machine {platform_machine}"
+
+        combined_markers = [
+            f"({marker})"
+            for marker in [sys_platform, markers, platform_machine]
+            if marker
+        ]
+
+        return " and ".join(combined_markers).strip()
+
+    def _fold_markers(self, dependency_tree, install_req):
+        comes_from = dependency_tree[install_req.name]
+
+        if comes_from == "Pipfile":
+            pipfile_entry = self.pipfile_entries.get(install_req.name)
+            if pipfile_entry and isinstance(pipfile_entry, dict):
+                return self._get_pipfile_markers(pipfile_entry)
+        else:
+            markers = self._fold_markers(dependency_tree, comes_from)
+            if markers:
```

Comment:
```suggestion
            if markers := self._fold_markers(dependency_tree, comes_from):
```

Reply:
I guess we can use the walrus operator, but it won't work in 3.7 so we essentially can hold this back until we drop 3.7, but I have mixed feelings about it with the open reports about hash collection against some private indexes.  🤔

Reply:
I am fine to remain Walrus-less.  My two freelance clients are both Python 3.11-only so I forgot about 3.7 in this case.